subprojects/junit-jupiter/src/main/java/org/mockito/junit/jupiter/MockitoExtension.java

```diff
@@ -160,7 +165,7 @@ public void beforeEach(final ExtensionContext context) {
         Set<Object> testInstances = new LinkedHashSet<>();
         testInstances.add(context.getRequiredTestInstance());
 
-        this.collectParentTestInstances(context, testInstances);
+        this.collectParentTestInstances(testInstances);
```

Comment:
This is not necessary anymore. Since you're on 5.4.2, you can remove the `ThreadLocal` and simply call

```java
List<Object> testInstances = context.getRequiredTestInstances().getAllInstances();
```

Reply:
Fantastic code review feedback!!! Many thanks!!!

Reply:
Also, thank you for this sweet new API :-)

Reply:
You're welcome! 😊